ibis/expr/types/generic.py

```diff
@@ -112,6 +112,39 @@ def cast(self, target_type: dt.DataType) -> Value:
 
         return op.to_expr()
 
+    def try_cast(self, target_type: dt.DataType) -> Value:
+        """Try cast expression to indicated data type.
+        If the cast fails for a row, the value is returned
+        as null or NaN depending on target_type and backend behavior.
```

Comment:
Are the backends besides `pandas` and `dask` that do this?  If not, I think we should probably call those out specifically (if the others all do the same-ish thing)

Reply:
I think I saw different behavior between clickhouse/trino/polars/duckdb but need to double check